### PR TITLE
[ADP-3486] Add a simple bash script to generate random deposit payments

### PR DESCRIPTION
**Pull request by abailly**
See the script's usage for details. This PR is in draft because I think the script is wrong: The wallet is not able to detect payment transactions it's making.

### CODE REVIEW

File: scripts/tx-gen/gen-deposits.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+  fi
+}
+
+for r in $random_u32; do
+  # account id between 0 and 100000
+  acct_id=$((r % 10000 ))
+
+  # derive public address
+  account_address=$(cat root.prv | cardano-address key child "1852H/1815H/${acct_id}H/0/0" | cardano-address key public --with-chain-code | cardano-address address payment --network-tag preprod)
```
Comment (Anviking) [inline scripts/tx-gen/gen-deposits.sh]: I believe it should be

```suggestion
  account_address=$(cat root.prv | cardano-address key child "1857H/1815H/0H/0/${addr_id}" | cardano-address key public --with-chain-code | cardano-address address payment --network-tag preprod)
```

based off 

```
deriveAddress s ix = rootXPrv s / 1857' / 1815' / 0' / 0 / ix
```

from https://github.com/cardano-foundation/cardano-wallet-agda/blob/43d1f83e4af85d318dcdbd912c8620bd05b60f78/lib/customer-deposit-wallet-pure/agda/Specification.lagda.md#address-derivation

**Comment by abailly** [comment]
So I was able to run the script and generating a transaction with 22 outputs paying to a random set of accounts between 0 and 10. The old wallet API shows the transaction correctly but the deposit wallet's API does not. /cc @paolino

**Comment by abailly** [comment]
Following changes introduced in #4863 we were able to observe deposit transactions' outputs in the wallet's UI which means the deposit wallet's address derivation matches cardano-addresses'

**Comment by abailly** [comment]
There's now a `gen-multiple-txs.sh` script that uses the `gen-deposits.sh` script to run a loop submitting transactions at a random interval. We have used it to populate the deposit wallet: 

Screenshot 2024-12-02 at 10 24 58

**Review by paolino — APPROVED** [review]
Thanks

**Pull request merged**